.gitignore

```diff
@@ -34,6 +34,7 @@ bin
 
 # Ignore files for local deployment
 config/overlays/development/manager_image_patch.yaml
+config/overlays/dev-image-config/inferenceservice_patch.yaml
```

Comment:
Any reason why this uses a different overlay ? Let's say I am developing PyTorch model server and want to add a new field `ModelClassParameters` field which need to change both KFServing controller and PyTorch model server, how the local development process looks like ?

Reply:
In this case, the local development process would be run ```make deploy-dev``` and  ```make  deploy-dev-pytorch``` with any order.
The reason why use a different overlay is that I think usually develop controller on go side and develop python on model sever side is not irrelevant, kfserving developer may modify go code but the end user may usually modify model server to fit for their use.  And also I don't want to make them all together because it does not make sense to build all of them even if only little changes did to just one component, this can speed up build too.

Reply:
That make sense but since they are different overlays wouldn’t they overwrite each other when they get applied in whatever order? If I do make deploy-dev first and then deploy-dev-pytorch, the later would overwrite previous deployed controller development image, no?

Reply:
Yes you are right, it's a loose end that I did not considered, need thing more about it... Do you have any suggestions?

Reply:
Think about it again, I in fact considered it for model servers in this PR but not for controller and logger,  I mentioned in development guide 
```
It's just for development and testing purpose, so in configmap, for predictors it will just keep the one in development, for others will just change the item impacted and set all others to be default. 
```
The key point is that we just kustomize build the static default version of yaml but not based on the current state in system. Even we build them all it may still be different from the running one in system since user may modify it through other method. Maybe it's a limitation unless we change the whole story to be based on the current state in k8s. Then it will be too heavy...

Reply:
So I think there's 2 ways to handle this:
- Still stick to the one by one rule and mention in development guide that the controller and logger will be set default too.
- Add some new make target like "deploy-dev-with-pytorch" and reorganize kustomize to build go side and pytorch together
Personally I think just mention it is good enough, because for the second case the combination would be much. Now that "deploy-dev-with-pytorch" and "deploy-dev-with-sklearn" is provided, why do we not provide "deploy-dev-with-pytorch_and_sklearn"

Reply:
I asked this because that is the real development use case I am having, often the time you need to change both controller and model server so I was hoping this can solve the problem. I think deploy-dev-sklearn make sense, we can explicitly say we do not support deploying controller with multiple model servers.

Reply:
Sure, updated development guide to mention that kfserving-controller-manager and logger  will also be set to default.

Reply:
Is there other comments on this? Or can we let this PR go, thanks.

Reply:
Sorry for the delay, this is great improvement for local python image development! We can have people try out first.